test/DGmethods_old/Euler/RTB_1DIMEX.jl

```diff
@@ -0,0 +1,427 @@
+using MPI
+using CLIMA
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.DGBalanceLawDiscretizations
+using CLIMA.DGBalanceLawDiscretizations1D
+using CLIMA.DGBalanceLawDiscretizations.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using CLIMA.Vtk
+using CLIMA.LinearSolvers
+using CLIMA.GeneralizedConjugateResidualSolver
+using CLIMA.AdditiveRungeKuttaMethod
+
+const γ_exact = 7 // 5 # FIXME: Remove this for some moist thermo approach
+
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters: R_d, cp_d, grav, cv_d, MSLP, T_0
+
+if haspkg("CuArrays")
+  using CUDAdrv
+  using CUDAnative
+  using CuArrays
+  CuArrays.allowscalar(false)
+  const ArrayType = CuArray
+else
+  const ArrayType = Array
+end
+
+const _nstate = 5
+const _δρ, _ρu, _ρv, _ρw, _δρe = 1:_nstate
+const _ρu⃗ = SVector(_ρu, _ρv, _ρw)
+const statenames = ("δρ", "ρu", "ρv", "ρw", "δρe")
+
+const _nauxstate = 6
+const _a_ρ0, _a_ρe0, _a_ϕ, _a_ϕ_x, _a_ϕ_y, _a_ϕ_z = 1:_nauxstate
+function auxiliary_state_initialization!(aux, x, y, z) #JK, dx, dy, dz)
+  @inbounds begin
+    ρ0, ρe0 = reference_ρ_ρe(x, y, z)
+    aux[_a_ρ0] = ρ0
+    aux[_a_ρe0] = ρe0
+    aux[_a_ϕ] = y * grav
+    aux[_a_ϕ_x] = 0
+    aux[_a_ϕ_y] = grav
+    aux[_a_ϕ_z] = 0
+  end
+end
+
+@inline function pressure(Q, aux)
+  @inbounds begin
+    gravity::eltype(Q) = grav
+    γ::eltype(Q) = γ_exact # FIXME: Remove this for some moist thermo approach
+    δρ, δρe = Q[_δρ], Q[_δρe]
+    ρu⃗ = SVector(Q[_ρu], Q[_ρv], Q[_ρw])
+    ρ0, ρe0, ϕ = aux[_a_ρ0], aux[_a_ρe0], aux[_a_ϕ]
+    ρ = ρ0 + δρ
+    ρe = ρe0 + δρe
+    ρinv = 1 / ρ
+    (γ-1)*(ρe - ρinv * (ρu⃗' * ρu⃗) / 2 - ϕ * ρ)
+  end
+end
+
+@inline function wavespeed(n, Q, aux, t)
+  γ::eltype(Q) = γ_exact # FIXME: Remove this for some moist thermo approach
+  n⃗ = SVector(n)
+  @inbounds begin
+    P = pressure(Q, aux)
+    δρ, δρe = Q[_δρ], Q[_δρe]
+    ρu⃗ = SVector(Q[_ρu], Q[_ρv], Q[_ρw])
+    ρ0, ρe0 = aux[_a_ρ0], aux[_a_ρe0]
+    ρ = ρ0 + δρ
+    ρe = ρe0 + δρe
+    ρinv = 1 / ρ
+    u⃗ = ρinv * ρu⃗
+    abs(n⃗' * u⃗) + sqrt(ρinv * γ * P)
```

Comment:
The speed of sound here and elsewhere is just an approximation, valid in the unstratified case. In the presence of stratification, there are (usually small) correction terms that depend, e.g., on the buoyancy frequency squared. (They are responsible for noise, e.g., from freeways being refracted at a stable inversion in winter. People in Pasadena experience this...) How important is it that the sound speed is accurate here?

Reply:
This is an approximate Riemann solver, like the ones used in finite volume methods.  Usually, we just want the maximum speed.  That could be locally or globally.  If it is too large it can add more dissipation than is needed.  But I don't think it needs to be extremely accurate.

Reply:
I agree with @lcw .  We just need an upper bound on the wave speed.

Reply:
Well. the Laplace wavespeed we use here is not the upper bound. It's probably good to within a few percent. Maybe that's good enough based on what @lcw says?